File: dayFour/java/TestAccountFunctions.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertArrayEquals;

public class TestAccountFunctions {
    
    @Test
    public void testThatTheBalanceIsZero() {
        double balance = 0.0;
        double expected = AccountFunctions.checkBalance(balance);
        double actual = 0.0;
        
        assertEquals(expected, actual);
    }
    @Test
    public void testThatDepositedAmountIsEqualToBalance(){
        double balance = 0.0;
        double depositAmount = 5000;
        balance = AccountFunctions.checkBalance(balance);
        double expected = AccountFunctions.deposit(depositAmount,balance, "");
        double actual = 5000;
        
        assertEquals(expected, actual);
    }
    @Test
    public void testThatWithdrawIsDeductedFromTheUpdatedBalance() {
        double withdrawAmount = 2000;
        double balance = AccountFunctions.deposit(5000, 0.0, "" );
        double expected = AccountFunctions.withdraw(withdrawAmount, balance, "");
        double actual = 3000;
        
        assertEquals(expected, actual);
    }
    @Test
    public void testThatWithdrawalIsNotGreaterThanBalance() {
        double withdrawAmount = 5000;
        double balance = 4000;
        balance = AccountFunctions.checkBalance(balance);
        double expected = AccountFunctions.withdraw(withdrawAmount, balance, "");
        double actual = AccountFunctions.checkBalance(balance);
        
        assertEquals(expected, actual);
    }
    @Test
    public void testThatTransactionsIsStoredInAllTransactionsArray() {
        double depositAmount = 5000;
        double balance = 0.0;
        balance = AccountFunctions.deposit(depositAmount, balance, "" );
        balance = AccountFunctions.checkBalance(balance);
        String transactions = "Deposited: ₦"; // + depositAmount + "| New Balance: ₦" + balance;
        
        String expected = AccountFunctions.showTransactions(transactions);
        String actual = transactions;
        
        assertEquals(expected, actual);
    }    
}

File: dayFour/python/test_account_functions.py


from unittest import TestCase

from account_functions import *

class Testaccount_functions(TestCase) :

    def testThatTheBalanceIsZero(self) :
        balance = 0.0
        actual = checkBalance(balance)
        expected = 0.0

        self.assertEqual(expected, actual)
    
    
    def testThatDepositedAmountIsEqualToBalance(self):
        balance = 0.0
        depositAmount = 5000
        balance = checkBalance(balance)
        actual = deposit(depositAmount,balance, "")
        expected = 5000

        self.assertEqual(expected, actual)
    
    
    def testThatWithdrawIsDeductedFromTheUpdatedBalance(self) :
        withdrawAmount = 2000
        balance = deposit(5000, 0.0, "" )
        actual = withdraw(withdrawAmount, balance, "")
        expected = 3000

        self.assertEqual(expected, actual)
    
    def testThatTheUserWithdrawedTwiceFromTheUpdatedBalance(self) :
        withdrawAmount = 2000
        balance = deposit(5000, 0.0, "" )
        first_withdraw = withdraw(withdrawAmount, balance, "")
        balance = checkBalance(first_withdraw)
        second_withdraw = withdraw(1000, balance, "")
 
        actual = checkBalance(second_withdraw)
        expected = 2000

        self.assertEqual(expected, actual)
    
    
    def testThatWithdrawalIsNotGreaterThanBalance(self) :
        withdrawAmount = 5000
        balance = 4000
        balance = checkBalance(balance)
        actual = withdraw(withdrawAmount, balance, "")
        expected = balance
        self.assertEqual(expected, actual)
        
    
    def testThatTheUserDepositedTwice(self) :
        balance = 0.0
        depositAmount = 5000;
        balance = checkBalance(balance)
        first_deposit = deposit(depositAmount,balance, "")
        balance = checkBalance(first_deposit)
        second_deposit = deposit(5000,balance, "")
        actual = checkBalance(second_deposit)
        expected = 10000
        self.assertEqual(expected, actual)
        
        
    def testThatTransactionsIsStoredInAllTransactionsArray(self) :
        depositAmount = 5000
        balance = 0.0
        balance = deposit(depositAmount, balance, "" )
        balance = checkBalance(balance)
        transactions = "Deposited: ₦" # + depositAmount + "| New Balance: ₦" + balance

        actual = showTransactions(transactions)
        expected = transactions

        self.assertEqual(expected, actual)
        
    
